### PR TITLE
Issue/23

### LINKED ISSUE #23: sol003-lifecycle-driver-vulnerabilities-issues
We have found the sol003-lifecycledriver-vulnerabilities given below CVE's.

**CVE-2021-33037** 
                 Apache Tomcat 10.0.0-M1 to 10.0.6, 9.0.0.M1 to 9.0.46 and 8.5.0 to 8.5.66 did not correctly parse the HTTP transfer-encoding request header in some.

**CVE-2021-30640**
                 A vulnerability in the JNDI Realm of Apache Tomcat allows an attacker to authenticate using variations of a valid user name and/or to bypass some of the protection provided by the LockOut Realm. This issue affects Apache Tomcat 10.0.0-M1 to 10.0.5; 9.0.0.M1 to 9.0.45; 8.5.0 to 8.5.65.

**CVE-2021-41079**
                 Apache Tomcat 8.5.0 to 8.5.63, 9.0.0-M1 to 9.0.43 and 10.0.0-M1 to 10.0.2 did not properly validate incoming TLS packets. When Tomcat was configured to use NIO+OpenSSL or NIO2+OpenSSL for TLS, a specially crafted packet could be used to trigger an infinite loop resulting in a denial of service.

**CVE-2021-42340​**
                 The fix for bug 63362 present in Apache Tomcat 10.1.0-M1 to 10.1.0-M5, 10.0.0-M1 to 10.0.11, 9.0.40 to 9.0.53 and 8.5.60 to 8.5.71 introduced a memory leak. The object introduced to collect metrics for HTTP upgrade connections was not released for WebSocket connections once the connection was closed. This created a memory leak that, over time, could lead to a denial of service via an OutOfMemoryError.

**CVE-2021-29425**
                  In Apache Commons IO before 2.7, When invoking the method FileNameUtils.normalize with an improper input string, like '//../foo', or '\\..\\foo', the result would be the same value, thus possibly providing access to files in the parent directory, but not further above (thus 'limited' path traversal), if the calling code would use the result to construct a path value.

**CVE-2020-9488**
                 Improper validation of certificate with host mismatch in Apache Log4j SMTP appender. This could allow an SMTPS connection to be intercepted by a man-in-the-middle attack which could leak any log messages sent through that appender.

**PRISMA-2021-0055**
                Versions <1.13 of this package are vulnerable to Information Exposure. When there is no byte array value that can be encoded into a string, the Base32 implementation does not reject it, and instead decodes it into an arbitrary value which can be re-encoded again using the same implementation. This allows for information exposure exploits such as tunneling additional information via seemingly valid base 32 strings. 

**CVE-2019-17495**
                 A Cascading Style Sheets (CSS) injection vulnerability in Swagger UI before 3.23.11 allows attackers to use the Relative Path Overwrite (RPO) technique to perform.

**CVE-2020-8908**
                 A temp directory creation vulnerability exists in all versions of Guava, allowing an attacker with access to the machine to potentially access data in a temporary directory created by the Guava API com.google.common.io.Files.createTempDir(). By default, on unix-like systems, the created directory is world-readable (readable by an attacker with access to the system). The method in question has been marked @Deprecated in versions 30.0 and later and should not be used. For Android developers, we recommend choosing a temporary directory API provided by Android, such as context.getCacheDir(). For other Java developers, we recommend migrating to the Java 7 API java.nio.file.Files.createTempDirectory() which explicitly configures permissions of 700, or configuring the Java runtime\'s java.io.tmpdir system property to point to a location whose permissions are appropriately configured.

**CVE-2018-10237**
                 Unbounded memory allocation in Google Guava 11.0 through 24.x before 24.1.1 allows remote attackers to conduct denial of service attacks against servers that depend on this library and deserialize attacker-provided data, because the AtomicDoubleArray class (when serialized with Java serialization) and the CompoundOrdering class (when serialized with GWT serialization) perform eager allocation without appropriate checks on what a client has sent and whether the data size is reasonable.

**CVE-2020-13956**
                 Apache HttpClient versions prior to version 4.5.13 and 5.0.3 can misinterpret malformed authority component in request URIs passed to the library as java.net.URI object and pick the wrong target host for request execution.

### PR DESCRIPTION
**Pull request by settipalli2F51**
Fixes #23

### CODE REVIEW

**Comment by settipalli2F51** [comment]
Hi @swatij-ibm @dvaccarosenna ,

Please assign reviewers for this PR  

Build Report : 
image

Cluster Installation Report  : 

image (1)

File: pom.xml [hunk 1]
```diff
@@ -29,20 +29,25 @@
...
         <helm.repo.snapshots.url>${helm.repo.addr}/accanto/snapshots</helm.repo.snapshots.url>
         <helm.repo.url>${helm.repo.addr}/accanto/stable</helm.repo.url>
+        <httpclient.version>4.5.13</httpclient.version>
+        <jackson.version>2.13.1</jackson.version>
         <jacoco-maven-plugin.version>0.8.4</jacoco-maven-plugin.version>
         <java.version>1.8</java.version>
-        <kafka.version>2.2.1</kafka.version>
+        <kafka.version>2.4.1</kafka.version>
         <kiwigrid.version>2.5</kiwigrid.version>
+        <log4j-api.version>2.13.2</log4j-api.version>
```
Comment (clloydaccanto) [inline pom.xml]: was there a reason log4j-api hasn't also been raised to 2.13.2? Would it make sense to match it up to the core version?

Comment (settipalli2F51) [reply pom.xml]: Hi @clloydaccanto 

Updated Log4j-api & Log4j-core versions  to 2.17.1 .

The Build is Successful and twist locks scan report is also doesn't show any additional  vulnerabilities .

Comment (settipalli2F51) [reply pom.xml]: Hi @clloydaccanto 

Latest Build Details : 

image

Cluster Installation Details 
image (2)

**Review by clloydaccanto — APPROVED** [review]
Thanks for the updates

**Review by swatij-ibm — APPROVED** [review]

**Review by hhellyer — APPROVED** [review]
LGTM - It looks like the issues identified in the reviews have been resolved.

**Pull request merged**